pkg/interfaces/contracts/vault/IAddUnbalancedLiquidityViaSwapRouter.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IERC4626 } from "@openzeppelin/contracts/interfaces/IERC4626.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IVault } from "./IVault.sol";
+import { IRouterQueries } from "./IRouterQueries.sol";
+
+import { AddLiquidityKind, RemoveLiquidityKind, SwapKind } from "./VaultTypes.sol";
+import "./RouterTypes.sol";
+
+/// @notice Router interface for adding unbalanced liquidity via a combination of a proportional add and a swap.
+interface IAddUnbalancedLiquidityViaSwapRouter {
+    struct AddLiquidityProportionalParams {
+        uint256[] maxAmountsIn;
+        uint256 exactBptAmountOut;
+        bytes userData;
+    }
+
+    struct SwapExactInParams {
```

Comment:
In the main contract you're using a combined struct. Consider defining that up here and using one struct instead of three. I tried it; it's not a big change and works.

Reply:
I mean you have SwapParams defined in the `AddUnbalancedLiquidityViaSwapRouter` contract. You could move that definition up to the interface `AddUnbalancedLiquidityViaSwapRouter`, and use it instead of defining the two structs: `SwapExactInParams` and `SwapExactOutParams`. The functions would still be separate; you're just using one struct instead of three.